flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitGenerator.java

```diff
@@ -22,25 +22,36 @@
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.util.List;
+import java.util.stream.IntStream;
 import org.apache.iceberg.CombinedScanTask;
 import org.apache.iceberg.Table;
 import org.apache.iceberg.TableProperties;
 import org.apache.iceberg.TableScan;
 import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.hadoop.Util;
 import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
 class FlinkSplitGenerator {
   private FlinkSplitGenerator() {
   }
 
-  static FlinkInputSplit[] createInputSplits(Table table, ScanContext context) {
+  static FlinkInputSplit[] createInputSplits(Table table, ScanContext context, boolean localityPreferred) {
     List<CombinedScanTask> tasks = tasks(table, context);
-    FlinkInputSplit[] splits = new FlinkInputSplit[tasks.size()];
-    for (int i = 0; i < tasks.size(); i++) {
-      splits[i] = new FlinkInputSplit(i, tasks.get(i));
-    }
-    return splits;
+
+    return IntStream.range(0, tasks.size()).parallel().mapToObj(i -> {
```

Comment:
Please use `Tasks` for parallelism like the Spark code, not Java parallelism. That's much more reliable.

Reply:
I read the Tasks, and it was well-designed and so impressive. Thanks for guidance.